components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/ServiceProvider.java

```diff
@@ -47,7 +47,15 @@
     private ClaimConfig claimConfig;
     private PermissionsAndRoleConfig permissionAndRoleConfig;
     private boolean saasApp;
+    private boolean idDumbMode = false;
```

Comment:
This can't be part of ServiceProvider. It should be part of InboundProvisioningConfig. Only "Basic Application Information" variables are directly under ServiceProvider. All other variables should go under the relevant categories.